### PR TITLE
[doit] Automatically use full path for prefix and create directory

### PR DESCRIPTION
**Pull request by doug65536**
Fix error

> configure: error: expected an absolute directory name for --prefix: ../build/x86_64-elf-elf-7.2.0-Linux-x86_64

### CODE REVIEW

File: doit [hunk 1]
```diff
@@ -135,6 +139,8 @@ fi
 
 if [ -z "$OUTDIR" ]; then
     OUTDIR=`pwd`
+else
+	mkdir -p "$OUTDIR"
```
[the remark this answers is not in the document]
Comment (doug65536) [reply doit]: Yes, that is a problem, all components of `readlink -f` must exist already. There is `readlink -m` but it is safer to defer the getting the full path until after the mkdir. I also forgot to check for `mkdir` error.

File: doit [hunk 2]
```diff
@@ -50,6 +50,10 @@ function log()
     fi
 }
 
+function full-path {
+    readlink -f "$1"
```
[the remark this answers is not in the document]
Comment (doug65536) [reply doit]: I realized that I don't need a canonical path, only an absolute path. Pushed new commit that makes the path absolute without running a program.

File: doit (hunk 1 above)
[the remark this answers is not in the document]
Comment (doug65536) [reply doit]: Pushed new commit that adds `--` and checks for `mkdir` error.

**Comment by travisg** [comment]
i rebased locally and merged

**Pull request closed without merging**